pkgs/native_toolchain_c/test/cbuilder/cbuilder_test.dart

```diff
@@ -19,112 +19,116 @@ import 'package:test/test.dart';
 import '../helpers.dart';
 
 void main() {
-  for (final buildMode in BuildMode.values) {
-    test('Cbuilder executable $buildMode', () async {
-      await inTempDir((tempUri) async {
-        final helloWorldCUri = packageUri
-            .resolve('test/cbuilder/testfiles/hello_world/src/hello_world.c');
-        if (!await File.fromUri(helloWorldCUri).exists()) {
-          throw Exception('Run the test from the root directory.');
-        }
-        const name = 'hello_world';
-
-        final buildConfig = BuildConfig(
-          outDir: tempUri,
-          packageRoot: tempUri,
-          targetArchitecture: Architecture.current,
-          targetOs: OS.current,
-          buildMode: buildMode,
-          // Ignored by executables.
-          linkModePreference: LinkModePreference.dynamic,
-          cCompiler: CCompilerConfig(
-            cc: cc,
-            envScript: envScript,
-            envScriptArgs: envScriptArgs,
-          ),
-        );
-        final buildOutput = BuildOutput();
-        final cbuilder = CBuilder.executable(
-          name: name,
-          sources: [helloWorldCUri.toFilePath()],
-        );
-        await cbuilder.run(
-          buildConfig: buildConfig,
-          buildOutput: buildOutput,
-          logger: logger,
-        );
-
-        final executableUri =
-            tempUri.resolve(Target.current.os.executableFileName(name));
-        expect(await File.fromUri(executableUri).exists(), true);
-        final result = await runProcess(
-          executable: executableUri,
-          logger: logger,
-        );
-        expect(result.exitCode, 0);
-        if (buildMode == BuildMode.debug) {
-          expect(result.stdout.trim(), startsWith('Running in debug mode.'));
-        }
-        expect(result.stdout.trim(), endsWith('Hello world.'));
-      });
-    });
-  }
-
-  for (final dryRun in [true, false]) {
-    final testSuffix = dryRun ? ' dry_run' : '';
-    test('Cbuilder dylib$testSuffix', () async {
-      await inTempDir(
-        // https://github.com/dart-lang/sdk/issues/40159
-        keepTemp: Platform.isWindows,
-        (tempUri) async {
-          final addCUri =
-              packageUri.resolve('test/cbuilder/testfiles/add/src/add.c');
-          const name = 'add';
+  for (final pic in [null, true, false]) {
+    for (final buildMode in BuildMode.values) {
+      test('Cbuilder executable $buildMode (pic: $pic)', () async {
```

Comment:
Please make a testSuffix here with buildMode and pic, that scales better if/when we add even more surrounding for loops.

Reply:
Done